benchexec/resources.py

```diff
@@ -15,191 +10,157 @@
+                    if sibling != core:
+                        cleanList.append(sibling)
+        for element in cleanList:
+            siblings_of_core.pop(element)
+        # siblings_of_core will be added to hierarchy_levels list after sorting
+
+        # read & prepare mapping of cores to L3 cache
+        # cores_of_L3cache = 
+        # hierarchy_levels.append(core_of_L3cache)
+        
+        # read & prepare mapping of cores to NUMA region
+        cores_of_NUMA_Region = get_NUMA_mapping (allCpus_list)
+        hierarchy_levels.append(cores_of_NUMA_Region)
+
+        # read & prepare mapping of cores to group
+        # core_of_group =
+        #hierarchy_levels.append(core_of_group)
+
+        # read & prepare mapping of cores to CPU/physical package/socket?
+        cores_of_package = get_package_mapping (allCpus_list)
+        hierarchy_levels.append(cores_of_package)
+    
     except ValueError as e:
         sys.exit(f"Could not read CPU information from kernel: {e}")
-    return _get_cpu_cores_per_run0(
-        coreLimit,
-        num_of_threads,
-        use_hyperthreading,
+    
+    # generate sorted list of dicts in accordance with their hierarchy
+    def compare_hierarchy(dict1, dict2):
+        value1 = len(next(iter(dict1.values())))
+        value2 = len(next(iter(dict2.values())))
+        if value1 > value2:
+            return 1
+        if value1 < value2:
+            return -1
+        if value1 == value2:
+            return 0
+
+    # sort hierarchy_levels according to the dicts' corresponding unit sizes
+    hierarchy_levels.sort(key = cmp_to_key(compare_hierarchy))  #hierarchy_level = [dict1, dict2, dict3]
+    # add siblings_of_core at the beginning of the list
+    hierarchy_levels.insert(0, siblings_of_core)
+
+    # create v_cores
+    allCpus = {}
+    for cpu_nr in allCpus_list:
+        allCpus.update({cpu_nr: v_core(cpu_nr,[])})
+
+    for level in hierarchy_levels:                          # hierarchy_levels = [dict1, dict2, dict3]
+        for key in level:
+            for core in level[key]:
+                allCpus[core].memory_regions.append(key)    # memory_regions = [key1, key2, key3]
+
+
+    # call the actual assignment function
+    return assignmentAlgorithm (
+        coreLimit, 
+        num_of_threads, 
+        use_hyperthreading, 
         allCpus,
-        cores_of_unit,
         siblings_of_core,
+        hierarchy_levels
     )
 
-
-def _get_cpu_cores_per_run0(
-    coreLimit,
-    num_of_threads,
-    use_hyperthreading,
+# define class v_core to generate core objects
+class v_core: 
+    #def __init__(self, id, siblings=None, memory_regions=None):
+    def __init__(self, id, memory_regions=None):
+        self.id = id
+        #self.siblings = siblings
+        self.memory_regions = memory_regions
+    def __str__(self):
+        return str(self.id) + " " + str(self.memory_regions)
+
+# assigns the v_cores into specific runs
+def assignmentAlgorithm (
```

Comment:
At many times, it is best if a function name contains a verb because it should describe what the function *does*. Can you please ensure this?